source/developers/plugins/studio-plugins.rst

```diff
@@ -0,0 +1,158 @@
+Creating a Crafter Studio Plugin
+--------------------------------
+Let'a take a look at how to create a Crafter Studio plugin.
+
+#. Create your plugin e.g. a JavaScript file or React app
+#. Create the required directory structure as outlined above then add your plug-in to your site under the ``config/studio/plugins/{yourPluginType}/{yourPluginName}`` directory
+
+      .. code-block:: text
+         :linenos:
+
+         {siteRoot}/
+           config/
+             studio/
+               plugins/
+                 {yourPluginType}/
+                   {yourPluginName}/
+
+      |
+
+#. Commit the new files added so it will be picked up by Studio
+#. If your plugin is inside Studio, setup needed configuration files, etc.
+#. See your plugin in action by refreshing your Studio browser if your plugin is inside Studio, otherwise visit: ``/studio/plugin?site={site}&type={yourPluginType}&name={yourPluginName}``
+
+.. note::
+
+   Here are some things to keep in mind when creating your full screen plugins with its own route:
+
+   - If your entry file is not called ``index.js``, you must add ``&file={yourFile}`` to the above url to see your plugin in action
+   - The steps listed above will load your plugin in the page. Your plugin would need to bootstrap and do whatever it needs to do when loaded i.e.  it should render itself and for that, it may need to create a root element and append it to the body.
+   - Some of our components, services and utils — including the ``AuthMonitor`` — are published via the ``CrafterCMSNext`` (window.CrafterCMSNext) global variable. This means you could use them in your plugin.
+
+
+-----------------------------------
+Using React To Develop Your Plugins
+-----------------------------------
+
+React is already present in the Studio client runtime. You may access the lib(s) via ``CrafterCMSNext`` (window.CrafterCMSNext).
+
+.. code-block:: js
+   :linenos:
+   :emphasize-lines: 3, 7, 8
+
+   (function () {
+
+     const { React, ReactDOM } = CrafterCMSNext;
+
+     CStudioAuthoring.Module.moduleLoaded('react-sample', {
+       initialize(config) {
+         ReactDOM.render(
+           React.createElement(
+             'div',
+             {
+               style: { margin: '10px 0' },
+               onClick() {
+                 console.log(config);
+               }
+             },
+             'Hello, this is a custom react plugin on the sidebar. ' +
+             'Click me to print my config values on the browser console.'
+           ),
+           config.containerEl
+         );
+       }
+     });
+
+   })();
+
+|
+
+
+You can use ``JSX``, ``TypeScript`` or any form of transpiling when developing your plugin. In this case, we suggest the following directory structure for your files:  ``{sandbox}/sources/{pluginSource}`` for the plugin source and ``{sandbox}/config/studio/plugins/{type}/{name}/(pluginBuild)`` for the JavaScript files/folders containing the plugin implementation
```

Comment:
May be more than just javascript. Let's just say the plugin build output or something more generic.